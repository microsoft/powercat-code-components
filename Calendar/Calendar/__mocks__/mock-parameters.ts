/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockDateTimeProperty, MockEnumProperty, MockStringProperty, MockTwoOptionsProperty } from './mock-context';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        SelectedDateValue: new MockDateTimeProperty(),
        MinDate: new MockDateTimeProperty(),
        MaxDate: new MockDateTimeProperty(),
        ShowGoToToday: new MockTwoOptionsProperty(),
        MonthPickerVisible: new MockTwoOptionsProperty(),
        DayPickerVisible: new MockTwoOptionsProperty(),
        HighlightSelectedMonth: new MockTwoOptionsProperty(),
        HighlightCurrentMonth: new MockTwoOptionsProperty(),
        ShowSixWeeksByDefault: new MockTwoOptionsProperty(),
        ShowWeekNumbers: new MockTwoOptionsProperty(),
        FirstDayOfWeek: new MockEnumProperty(),
        BackgroundColor: new MockStringProperty(),
    };
}
