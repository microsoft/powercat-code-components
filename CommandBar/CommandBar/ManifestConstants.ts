export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    DisplayName = 'ItemDisplayName',
    Key = 'ItemKey',
    Enabled = 'ItemEnabled',
    IconName = 'ItemIconName',
    IconColor = 'ItemIconColor',
    Overflow = 'ItemOverflow',
    FarItem = 'ItemFarItem',
    ParentKey = 'ItemParentKey',
    IconOnly = 'ItemIconOnly',
    Visible = 'ItemVisible',
    Checked = 'ItemChecked',
    Split = 'ItemSplit',
    ItemHeader = 'ItemHeader',
    ItemTopDivider = 'ItemTopDivider',
    ItemDivider = 'ItemDivider',
    ItemStyles = 'ItemStyles',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
}

export const enum InputProperties {
    InputEvent = 'InputEvent',
    SelectedKey = 'SelectedKey',
}
