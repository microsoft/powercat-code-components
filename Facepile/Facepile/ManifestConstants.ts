export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    DisplayName = 'ItemPersonaName',
    Key = 'ItemPersonaKey',
    ImageUrl = 'ItemPersonaImageUrl',
    Image = 'ItemPersonaImage',
    ImageInfo = 'ItemPersonaImageInfo',
    Role = 'ItemPersonaRole',
    ImageType = 'ItemImageType',
    Presence = 'ItemPersonaPresence',
    Clickable = 'ItemPersonaClickable',
    IsImage = 'IsImage',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
}

export const enum InputProperties {
    InputEvent = 'InputEvent',
    SelectedKey = 'SelectedKey',
}

export const enum OutputEvents {
    PersonaEvent = 'PersonaEvent',
    OverFlowButtonEvent = 'OverFlowButtonEvent',
    AddButtonEvent = 'AddButtonEvent',
}
