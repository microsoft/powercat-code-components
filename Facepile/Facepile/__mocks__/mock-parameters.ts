/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
        MaxDisplayablePersonas: new MockWholeNumberProperty(),
        ImageShouldFadeIn: new MockTwoOptionsProperty(),
        PersonaSize: new MockEnumProperty(),
        OverflowButtonType: new MockEnumProperty(),
        ShowAddButton: new MockTwoOptionsProperty(),
        OverflowButtonAriaLabel: new MockStringProperty(),
        AddbuttonAriaLabel: new MockStringProperty(),
    };
}
