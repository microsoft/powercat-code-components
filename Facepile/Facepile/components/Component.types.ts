import { PersonaSize, IFacepilePersona, OverflowButtonType } from '@fluentui/react';

export interface IFacepileprops {
    width?: number;
    height?: number;
    text?: string;
    textColor?: string;
    imagesFadeIn?: boolean;
    displayedPersonas?: number;
    personaSize: PersonaSize;
    items: ICustomFacepile[];
    disabled?: boolean;
    tabIndex?: number;
    ariaLabel?: string;
    themeJSON?: string;
    setFocus?: string;
    overflowButtonType?: OverflowButtonType;
    onSelected: (eventName: string, item?: IFacepilePersona) => void;
    showAddButton: boolean;
    overflowButtonAriaLabel: string;
    addbuttonAriaLabel: string;
}

export interface ICustomFacepile {
    id: string;
    key: string;
    personaName: string;
    imageUrl: string;
    presence: string;
    imageInitials?: string;
    initialsColor?: number;
    clickable: boolean;
    onClick?: (ev: unknown, persona?: IFacepilePersona) => void;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    data?: any;
}
