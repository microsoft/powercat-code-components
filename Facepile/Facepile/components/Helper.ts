import { ItemColumns } from '../ManifestConstants';
import { PersonaSize, OverflowButtonType, PersonaPresence } from '@fluentui/react';

export function getUrlfromImage(record: ComponentFramework.PropertyHelper.DataSetApi.EntityRecord): string {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    const imageData = undefinedIfNullish(record.getValue(ItemColumns.Image) as any) ?? '';
    return imageData ? `data:image/jpeg;base64, ${imageData.fileContent}` : imageData;
}

export function getImageUrl(record: ComponentFramework.PropertyHelper.DataSetApi.EntityRecord): string {
    return (record.getValue(ItemColumns.ImageInfo) as string) ?? '';
}

export function undefinedIfNullish<T>(value: T): T | undefined {
    return defaultIfNullish(value, undefined);
}
function defaultIfNullish<T>(value: T, defaultValue: T) {
    return (value as T) ? value : defaultValue;
}

export function getPersonaPresence(record: ComponentFramework.PropertyHelper.DataSetApi.EntityRecord): PersonaPresence {
    const personaPresence = undefinedIfNullish(record.getValue(ItemColumns.Presence));
    if (personaPresence) {
        switch ((personaPresence as string).toLowerCase()) {
            case 'away':
                return PersonaPresence.away;
            case 'blocked':
                return PersonaPresence.blocked;
            case 'busy':
                return PersonaPresence.busy;
            case 'dnd':
                return PersonaPresence.dnd;
            case 'none':
                return PersonaPresence.none;
            case 'offline':
                return PersonaPresence.offline;
            case 'online':
                return PersonaPresence.online;
            default:
                return PersonaPresence.none;
        }
    } else {
        return PersonaPresence.none;
    }
}

export function getPersonaSize(sizeSelected: string): PersonaSize {
    switch (sizeSelected.toLowerCase()) {
        case 'size8':
            return PersonaSize.size8;
        case 'size24':
            return PersonaSize.size24;
        case 'size32':
            return PersonaSize.size32;
        case 'size40':
            return PersonaSize.size40;
        case 'size48':
            return PersonaSize.size48;
        case 'size56':
            return PersonaSize.size56;
        default:
            return PersonaSize.size32;
    }
}

export function getOverflowButtonType(btnTypeSelected: string): OverflowButtonType {
    switch (btnTypeSelected.toLowerCase()) {
        case 'none':
            return OverflowButtonType.none;
        case 'descriptive':
            return OverflowButtonType.descriptive;
        case 'downarrow':
            return OverflowButtonType.downArrow;
        case 'more':
            return OverflowButtonType.more;
        default:
            return OverflowButtonType.descriptive;
    }
}
