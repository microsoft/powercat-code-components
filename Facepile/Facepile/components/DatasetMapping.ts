import { IFacepilePersona } from '@fluentui/react';
import { ItemColumns } from '../ManifestConstants';
import { ICustomFacepile } from './Component.types';
import { undefinedIfNullish, getImageUrl, getUrlfromImage } from './Helper';

export function getitemFromDataset(dataset: ComponentFramework.PropertyTypes.DataSet): ICustomFacepile[] {
    if (dataset.error || dataset.paging.totalResultCount === undefined) {
        // Dataset is not defined so return dummy items
        return getDummyAction();
    }
    const keyIndex: Record<string, number> = {};
    return dataset.sortedRecordIds.map((id) => {
        const record = dataset.records[id];
        // Prevent duplicate keys by appending the duplicate index
        let key = record.getValue(ItemColumns.Key) as string;

        if (keyIndex[key] !== undefined) {
            keyIndex[key]++;
            key += `_${keyIndex[key]}`;
        } else keyIndex[key] = 1;
        return {
            id: record.getRecordId(),
            key: key,
            personaName: record.getFormattedValue(ItemColumns.DisplayName),
            imageUrl: undefinedIfNullish(record.getValue(ItemColumns.IsImage) as boolean)
                ? getUrlfromImage(record)
                : getImageUrl(record),
            data: record,
            presence: record.getValue(ItemColumns.Presence) as string,
            clickable: (record.getValue(ItemColumns.Clickable) as boolean) ?? false,
        } as ICustomFacepile;
    });
}

export function getFacepilePersonas(
    items: ICustomFacepile[],
    onClick: (ev?: unknown, item?: IFacepilePersona) => void,
): ICustomFacepile[] {
    return items.map((item: ICustomFacepile) => ({
        id: item.id,
        key: item.key,
        personaName: item.personaName,
        presence: item.presence,
        imageUrl: item.imageUrl,
        ...(item.clickable && { onClick: onClick }),
    })) as ICustomFacepile[];
}

function getDummyAction(): ICustomFacepile[] {
    return [
        {
            id: '1',
            key: '1',
            personaName: 'Megan Bowen',
            imageUrl:
                'https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/persona-female.png',
            imageInitials: 'Item ' + 1,
            initialsColor: 1,
            presence: 'none',
        },
        {
            id: '2',
            key: '2',
            personaName: 'Diego Siciliano',
            imageUrl:
                'https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/persona-male.png',
            imageInitials: 'Item ' + 1,
            initialsColor: 1,
            presence: 'none',
        },
    ] as ICustomFacepile[];
}
