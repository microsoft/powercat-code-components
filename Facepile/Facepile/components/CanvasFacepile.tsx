import * as React from 'react';
import { Facepile as CustomFacepile } from '../fluenui-fork/Facepile/Facepile';
import { IFacepilePersona, ThemeProvider, createTheme, IPartialTheme } from '@fluentui/react';
import { IFacepileprops } from './Component.types';
import { getFacepilePersonas } from './DatasetMapping';
import { getPersonaPresence } from './Helper';
import { useAsync } from '@fluentui/react-hooks';
import { OutputEvents } from '../ManifestConstants';

export const CanvasFacepile = React.memo((props: IFacepileprops) => {
    const {
        width,
        height,
        imagesFadeIn,
        displayedPersonas,
        onSelected,
        personaSize,
        setFocus,
        themeJSON,
        ariaLabel,
        items,
        overflowButtonType,
        showAddButton,
        overflowButtonAriaLabel,
        addbuttonAriaLabel,
        tabIndex,
    } = props;

    const theme = React.useMemo(() => {
        try {
            return themeJSON ? createTheme(JSON.parse(themeJSON) as IPartialTheme) : undefined;
        } catch (ex) {
            /* istanbul ignore next */
            console.error('Cannot parse theme', ex);
        }
    }, [themeJSON]);

    const getPersonaProps = React.useCallback(
        (persona: IFacepilePersona) => ({
            imageShouldFadeIn: imagesFadeIn,
            ...(typeof persona.data !== 'undefined' && { presence: getPersonaPresence(persona.data) }),
        }),
        [imagesFadeIn],
    );

    const overflowButtonProps = {
        ariaLabel: overflowButtonAriaLabel,
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        onClick: (ev: React.MouseEvent<HTMLButtonElement>) => onSelected(OutputEvents.OverFlowButtonEvent),
    };

    const addButtonProps = {
        ariaLabel: addbuttonAriaLabel,
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        onClick: (ev: React.MouseEvent<HTMLButtonElement>) => onSelected(OutputEvents.AddButtonEvent),
    };

    // Provide an explicit size so that the ResizeGroup measurements are correct
    const rootStyle = React.useMemo(() => {
        return {
            display: 'block',
            position: 'absolute',
            width: width,
            height: height,
        } as React.CSSProperties;
    }, [width, height]);

    const onClick = React.useCallback(
        (ev?: unknown, item?: IFacepilePersona) => {
            const selectedItem = item && items.find((i: IFacepilePersona) => i.id === item?.id);
            if (selectedItem) onSelected(OutputEvents.PersonaEvent, selectedItem);
        },
        [items, onSelected],
    );

    const rootRef = React.useRef<HTMLDivElement>(null);
    const async = useAsync();
    React.useEffect(() => {
        if (setFocus && setFocus !== '' && rootRef) {
            async.requestAnimationFrame(() => {
                // We can't call focus() on the imperative componentRef because of a bug in ResizeGroup
                // that causes the componentRef.current to be nulled
                // See https://github.com/microsoft/fluentui/issues/22844
                const buttons = (rootRef.current as HTMLElement).getElementsByTagName('button');
                if (buttons && buttons.length > 0) {
                    buttons[0].focus();
                }
            });
        }
    }, [setFocus, rootRef, async]);

    const facepilePersonas: IFacepilePersona[] = getFacepilePersonas(items, onClick);

    return (
        <ThemeProvider theme={theme} style={rootStyle}>
            <CustomFacepile
                personaSize={personaSize}
                personas={facepilePersonas}
                maxDisplayablePersonas={displayedPersonas}
                overflowButtonType={overflowButtonType}
                overflowButtonProps={overflowButtonProps}
                getPersonaProps={getPersonaProps}
                ariaLabel={ariaLabel}
                showAddButton={showAddButton}
                addButtonProps={addButtonProps}
                tabIndex={tabIndex}
            />
        </ThemeProvider>
    );
});
CanvasFacepile.displayName = 'CanvasFacepile';
