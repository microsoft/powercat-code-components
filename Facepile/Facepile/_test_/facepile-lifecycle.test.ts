import { Facepile } from '..';
import { IInputs } from '../generated/ManifestTypes';
import { ItemColumns, OutputEvents } from '../ManifestConstants';
import { MockContext, MockState } from '../__mocks__/mock-context';
import { MockDataSet, MockEntityRecord } from '../__mocks__/mock-datasets';
import { getMockParameters } from '../__mocks__/mock-parameters';
import { act } from 'react-dom/test-utils';
import { mount } from 'enzyme';
import { getPersonaPresence, getPersonaSize } from '../components/Helper';
import { PersonaPresence, PersonaSize } from '@fluentui/react';

// Since requestAnimationFrame does not exist in the test DOM, mock it
window.requestAnimationFrame = jest.fn().mockImplementation((callback) => {
    callback();
});

jest.useFakeTimers();

describe('FacePile', () => {
    beforeEach(() => jest.clearAllMocks());
    afterEach(() => {
        for (let i = 0; i < document.body.children.length; i++) {
            if (document.body.children[i].tagName === 'DIV') {
                document.body.removeChild(document.body.children[i]);
                i--;
            }
        }
    });

    it('renders', () => {
        const { component, context, notifyOutputChanged } = createComponent();
        component.init(context, notifyOutputChanged);
        const element = component.updateView(context);
        expect(element).toMatchSnapshot();
    });

    it('renders dummy items when no items configured', () => {
        const { component, context, notifyOutputChanged } = createComponent();
        // Simulate there being no items bound - which causes an error on the parameters
        context.parameters.items.error = true;
        component.init(context, notifyOutputChanged);
        const element = component.updateView(context);
        expect(element).toMatchSnapshot();
    });

    it('check getPersonaPresence', () => {
        let presence: string | undefined = 'none';
        let mockRecord = getMockDataSet(presence);
        expect(getPersonaPresence(mockRecord.records['1'])).toBe(PersonaPresence.none);
        presence = 'away';
        mockRecord = getMockDataSet(presence);
        expect(getPersonaPresence(mockRecord.records['1'])).toBe(PersonaPresence.away);
        presence = 'blocked';
        mockRecord = getMockDataSet(presence);
        expect(getPersonaPresence(mockRecord.records['1'])).toBe(PersonaPresence.blocked);
        presence = 'busy';
        mockRecord = getMockDataSet(presence);
        expect(getPersonaPresence(mockRecord.records['1'])).toBe(PersonaPresence.busy);
        presence = 'dnd';
        mockRecord = getMockDataSet(presence);
        expect(getPersonaPresence(mockRecord.records['1'])).toBe(PersonaPresence.dnd);
        presence = 'offline';
        mockRecord = getMockDataSet(presence);
        expect(getPersonaPresence(mockRecord.records['1'])).toBe(PersonaPresence.offline);
        presence = 'online';
        mockRecord = getMockDataSet(presence);
        expect(getPersonaPresence(mockRecord.records['1'])).toBe(PersonaPresence.online);
        presence = 'focused';
        mockRecord = getMockDataSet(presence);
        expect(getPersonaPresence(mockRecord.records['1'])).toBe(PersonaPresence.none);
        presence = '';
        mockRecord = getMockDataSet(presence);
        expect(getPersonaPresence(mockRecord.records['1'])).toBe(PersonaPresence.none);
    });

    it('check getPersonaSize', () => {
        expect(getPersonaSize('size8')).toBe(PersonaSize.size8);
        expect(getPersonaSize('size24')).toBe(PersonaSize.size24);
        expect(getPersonaSize('size32')).toBe(PersonaSize.size32);
        expect(getPersonaSize('size40')).toBe(PersonaSize.size40);
        expect(getPersonaSize('size48')).toBe(PersonaSize.size48);
        expect(getPersonaSize('size56')).toBe(PersonaSize.size56);
        expect(getPersonaSize('')).toBe(PersonaSize.size32);
    });

    it('check persona onclick event', () => {
        const { component, context, notifyOutputChanged } = createComponent();

        component.init(context, notifyOutputChanged);
        const facepileComponent = component.updateView(context);

        const firstCommandReference = {
            id: { guid: '1' },
            name: '1',
        } as ComponentFramework.EntityReference;

        context.parameters.items.records['1'].getNamedReference = jest.fn().mockReturnValueOnce(firstCommandReference);
        const facepile = mount(facepileComponent);
        const facepileNode = facepile.find('.ms-Facepile-itemButton').first();
        expect(facepileNode.length).toEqual(1);

        facepileNode.simulate('click');
        expect(context.parameters.items.openDatasetItem).toBeCalledTimes(1);
        expect(context.parameters.items.openDatasetItem).toBeCalledWith(firstCommandReference);
        const outputs = component.getOutputs();
        expect(outputs.EventName).toEqual(OutputEvents.PersonaEvent);
    });

    it('check output events', () => {
        const { component, context, notifyOutputChanged } = createComponent();
        component.init(context, notifyOutputChanged);
        context.parameters.ShowAddButton.raw = true;
        context.parameters.MaxDisplayablePersonas.raw = 1;
        const facepileComponent = component.updateView(context);

        const firstCommandReference = {
            id: { guid: '1' },
            name: '1',
        } as ComponentFramework.EntityReference;

        context.parameters.items.records['1'].getNamedReference = jest.fn().mockReturnValueOnce(firstCommandReference);
        const facepile = mount(facepileComponent);
        const addbtnElement = facepile.find('.ms-Facepile-addButton').first();
        expect(addbtnElement.length).toEqual(1);

        addbtnElement.simulate('click');
        let outputs = component.getOutputs();
        expect(outputs.EventName).toEqual(OutputEvents.AddButtonEvent);

        const overflowElement = facepile.find('.ms-Facepile-descriptiveOverflowButton').first();
        expect(overflowElement.length).toEqual(1);
        overflowElement.simulate('click');
        outputs = component.getOutputs();
        expect(outputs.EventName).toEqual(OutputEvents.OverFlowButtonEvent);
    });

    it('check overflowbutton behaviour', () => {
        const { component, context, notifyOutputChanged } = createComponent();
        component.init(context, notifyOutputChanged);
        context.parameters.ShowAddButton.raw = true;
        context.parameters.MaxDisplayablePersonas.raw = 1;
        let facepileComponent = component.updateView(context);
        let facepile = mount(facepileComponent);
        let overflowElement = facepile.find('.ms-Facepile-descriptiveOverflowButton').first();
        expect(overflowElement.length).toEqual(1);

        context.parameters.OverflowButtonType.raw = 'more';
        facepileComponent = component.updateView(context);
        facepile = mount(facepileComponent);
        overflowElement = facepile.find('.ms-Facepile-overflowButton').first();
        expect(overflowElement.length).toEqual(1);

        context.parameters.OverflowButtonType.raw = 'none';
        facepileComponent = component.updateView(context);
        facepile = mount(facepileComponent);
        overflowElement = facepile.find('.ms-Facepile-overflowButton').first();
        expect(overflowElement.length).toEqual(0);
    });

    it('theme', async () => {
        const { component, context, container, notifyOutputChanged } = createComponent();
        context.parameters.Theme.raw = JSON.stringify({
            palette: {
                themePrimary: '#0078d4',
            },
        });
        act(() => {
            component.init(context, notifyOutputChanged);
            component.updateView(context);
        });

        expect(container).toMatchSnapshot();
    });
});

function createComponent() {
    const component = new Facepile();
    const notifyOutputChanged = jest.fn();
    const context = new MockContext<IInputs>(getMockParameters());
    context.parameters.PersonaSize.raw = 'Size8';
    context.parameters.OverflowButtonType.raw = 'descriptive';
    context.parameters.ImageShouldFadeIn.raw = true;
    context.parameters.items = getMockDataSet('away');
    const state = new MockState();
    const container = document.createElement('div');
    document.body.appendChild(container);
    return { component, context, container, notifyOutputChanged, state };
}

function getMockDataSet(presence: string) {
    return new MockDataSet([
        new MockEntityRecord('1', {
            [ItemColumns.Key]: 'item1',
            [ItemColumns.DisplayName]: 'Diego Siciliani',
            [ItemColumns.ImageUrl]:
                'https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/persona-male.png',
            [ItemColumns.Clickable]: true,
            [ItemColumns.Presence]: presence,
            [ItemColumns.IsImage]: false,
        }),
        new MockEntityRecord('2', {
            [ItemColumns.Key]: 'item2',
            [ItemColumns.DisplayName]: 'Megan Bowen',
            [ItemColumns.ImageUrl]:
                'https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/persona-female.png',
            [ItemColumns.Clickable]: true,
            [ItemColumns.Presence]: presence,
            [ItemColumns.IsImage]: false,
        }),
    ]);
}
