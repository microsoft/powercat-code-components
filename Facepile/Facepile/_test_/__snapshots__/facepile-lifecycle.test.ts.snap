// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacePile renders 1`] = `
<Memo(CanvasFacepile)
  addbuttonAriaLabel=""
  ariaLabel=""
  displayedPersonas={5}
  height={-1}
  imagesFadeIn={true}
  items={
    Array [
      Object {
        "clickable": true,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "1",
          "values": Object {
            "IsImage": false,
            "ItemPersonaClickable": true,
            "ItemPersonaImageUrl": "https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/persona-male.png",
            "ItemPersonaKey": "item1",
            "ItemPersonaName": "Diego Siciliani",
            "ItemPersonaPresence": "away",
          },
        },
        "id": "1",
        "imageUrl": "",
        "key": "item1",
        "personaName": "Diego Siciliani",
        "presence": "away",
      },
      Object {
        "clickable": true,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "2",
          "values": Object {
            "IsImage": false,
            "ItemPersonaClickable": true,
            "ItemPersonaImageUrl": "https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/persona-female.png",
            "ItemPersonaKey": "item2",
            "ItemPersonaName": "Megan Bowen",
            "ItemPersonaPresence": "away",
          },
        },
        "id": "2",
        "imageUrl": "",
        "key": "item2",
        "personaName": "Megan Bowen",
        "presence": "away",
      },
    ]
  }
  onSelected={[Function]}
  overflowButtonAriaLabel=""
  overflowButtonType={1}
  personaSize={17}
  setFocus=""
  tabIndex={0}
  width={-1}
/>
`;

exports[`FacePile renders dummy items when no items configured 1`] = `
<Memo(CanvasFacepile)
  addbuttonAriaLabel=""
  ariaLabel=""
  displayedPersonas={5}
  height={-1}
  imagesFadeIn={true}
  items={
    Array [
      Object {
        "id": "1",
        "imageInitials": "Item 1",
        "imageUrl": "https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/persona-female.png",
        "initialsColor": 1,
        "key": "1",
        "personaName": "Megan Bowen",
        "presence": "none",
      },
      Object {
        "id": "2",
        "imageInitials": "Item 1",
        "imageUrl": "https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/persona-male.png",
        "initialsColor": 1,
        "key": "2",
        "personaName": "Diego Siciliano",
        "presence": "none",
      },
    ]
  }
  onSelected={[Function]}
  overflowButtonAriaLabel=""
  overflowButtonType={1}
  personaSize={17}
  setFocus=""
  tabIndex={0}
  width={-1}
/>
`;

exports[`FacePile theme 1`] = `<div />`;
